File: src/com/seancheey/gui/MainController.java
package com.seancheey.gui;

import javafx.application.Application;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.TextField;
import javafx.stage.Stage;

public class MainController extends Application {
    @FXML
    private TextField username, password;

    private static Stage PrimaryStage;

    @Override
    public void start(Stage primaryStage) throws Exception {
        Parent root = FXMLLoader.load(getClass().getResource("main.fxml"));
        primaryStage.setTitle("Amao Game");
        primaryStage.setScene(new Scene(root, 800, 600));
        PrimaryStage = primaryStage;
        primaryStage.show();
    }


    public static void main(String[] args) {
        launch(args);
    }

    public void login() throws Exception {
        System.out.println(username.getText() + "\n" + password.getText());
        Parent root = FXMLLoader.load(getClass().getResource("bot_edit.fxml"));
        PrimaryStage.setScene(new Scene(root, 800, 600));
    }

    public void register() {

    }

    public void setting() {

    }

    public void credit() {

    }
}

File: src/com/seancheey/gui/MainController.kt
package com.seancheey.gui

/**
 * Created by Seancheey on 27/05/2017.
 * GitHub: https://github.com/Seancheey
 */

import com.seancheey.game.Config
import com.seancheey.game.Player
import com.seancheey.game.PlayerSavesReader
import javafx.fxml.FXML
import javafx.scene.Scene
import javafx.scene.control.TextField


class MainController {
    @FXML
    private var username: TextField? = null
    @FXML
    private var password: TextField? = null

    fun login() {
        if (username!!.text != "") {
            username!!.text = "guest"
        }
        // id method will be changed later
        val id = username!!.text.toHashSet().hashCode().toLong()
        Config.player = PlayerSavesReader("saves/$id.object").readPlayer() ?: Player(id, username!!.text)

        Stages.primaryStage!!.scene = Scene(Scenes.menu, 800.0, 600.0)
    }

    fun register() {

    }

    fun setting() {

    }

    fun credit() {

    }
}
